pkg/ebpf/maps/generic_map.go

```diff
@@ -53,7 +55,18 @@ var BatchAPISupported = funcs.MemoizeNoError(func() bool {
 // GenericMap is a wrapper around ebpf.Map that allows to use generic types.
 // Also includes support for batch iterations
 type GenericMap[K any, V any] struct {
-	m *ebpf.Map
+	m                   *ebpf.Map
+	keySupportsBatchAPI bool
+}
+
+func canBinaryReadKey[K any]() bool {
+	kvalList := make([]K, 1)
```

Comment:
Does this need to be a slice, or could it be `kval := new(K)`?

Reply:
Changed